extended_tasks/mt_bench/main.py

```diff
@@ -0,0 +1,138 @@
+This file generally create just a TASKS_TABLE and TASKS_GROUPS which are then imported by LightEval.
+Author:
+"""
+
+import numpy as np
+from aenum import extend_enum
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from extended_tasks.mt_bench.judges import JudgeOpenAI
+from lighteval.metrics import Metrics
+from lighteval.metrics.utils import MetricCategory, MetricUseCase, SampleLevelMetric, SampleLevelMetricGrouping
+from lighteval.tasks.lighteval_task import LightevalTaskConfig
+from lighteval.tasks.requests import Doc
+from lighteval.tasks.tasks_prompt_formatting import LETTER_INDICES
+
+
+# EVAL WITH NO SUBSET ##
+# This is how you create a simple tasks (like hellaswag) which has one single subset
+# attached to it, and one evaluation possible.
+task = LightevalTaskConfig(
+    name="mt_bench",
+    prompt_function="prompt_fn",  # must be defined in the file or imported from src/lighteval/tasks/tasks_prompt_formatting.py
+    suite=["extended"],
+    hf_repo="SaylorTwift/mt-bench",
+    hf_subset="default",
+    hf_avail_splits=["train"],
+    evaluation_splits=["train"],
+    few_shots_split="",
+    few_shots_select="random",
+    metric=["mt_bench_metric"],
+    generation_size=1024,
+    stop_sequence=[],
+)
+
+
+# DEFINE YOUR PROMPT FUNCTIONS
+# Define as many as you need for your different tasks
+def prompt_fn(line, task_name: str = None):
+    """Defines how to go from a dataset line to a doc object.
+    Follow examples in src/lighteval/tasks/tasks_prompt_formatting.py, or get more info
+    about what this function should do in the README.
+    """
+    return Doc(
+        task_name=task_name,
+        query=f"{line['turns'][0]}",
+        choices=None,
+        instruction=None,
+        gold_index=[],
+        specific={
+            "reference": line["reference"],
+            "category": line["category"],
+            "multi_turn_queries": line["turns"],
+            "id": line["question_id"],
+        },
+    )
+
+
+def mt_bench_metric(predictions: list[str], formatted_doc: Doc, **kwargs) -> dict[str, float]:
+    """Defines how to go from a list of predictions to a score.
+    Follow examples in src/lighteval/metrics/metrics.py, or get more info
+    about what this function should do in the README.
+    """
+
+    judge = JudgeOpenAI(
+        model="gpt-3.5-turbo",
+        seed=42,
+        temperature=0.0,
+        templates_path="extended_tasks/mt_bench/judge_prompts.jsonl",
+    )
+
+    questions = formatted_doc.specific["multi_turn_queries"]
+    ref_answers = formatted_doc.specific["reference"]
+
+    score, messages, judgement = judge.evaluate_answer(questions, predictions, ref_answers, single_turn=True)
+    score_mt, messages_mt, judgement_mt = judge.evaluate_answer(questions, predictions, ref_answers, single_turn=False)
+
+    return {
+        "single_turn": score,
+        "multi_turn": score_mt,
+        "user_prompt": [messages, messages_mt],
```

Comment:
are you returning the user_prompt and judgement here for later logging? 
If yes, we need to look for a better system in the future, for example, by adding them to formatted_doc specific instead

Reply:
Yes, I was thinking of either adding them to the doc or having a logger in the metrics in case we want to log any other things for other metrics. wdyt ?

Reply:
Interesting idea to get a metric logger. Could be worth doing in the future